ateam_ai/src/ateam_ai_node.cpp

```diff
@@ -163,6 +170,57 @@ class ATeamAINode : public rclcpp::Node
     ball_state.vel.y() = ball_state_msg->twist.linear.y;
   }
 
+  void field_callback(const ssl_league_msgs::msg::VisionGeometryFieldSize::SharedPtr field_msg)
+  {
+
+    Field field {
+        .field_length   = field_msg->field_length,
+        .field_width    = field_msg->field_width,
+        .goal_width     = field_msg->goal_width,
+        .goal_depth     = field_msg->goal_depth,
+        .boundary_width = field_msg->boundary_width
+    };
+
+    auto check_field_line_name = [](ssl_league_msgs::msg::VisionFieldLineSegment line_msg, std::string target_name) -> bool {
+        return line_msg.name == target_name; 
+    };
+    std::array<std::string, 4> field_bound_names = {"TopTouchLine", "BottomTouchLine", "LeftGoalLine", "RightGoalLine"};
+    for (size_t i = 0; i < field_bound_names.size(); i++) {
+        auto& name = field_bound_names.at(i);
+        auto itr = std::find_if(begin(field_msg->field_lines), end(field_msg->field_lines), std::bind(check_field_line_name, std::placeholders::_1, name));
+        if (itr != end(field_msg->field_lines)) {
+            field.field_corners.at(i)(0) = itr->p1.x;
+            field.field_corners.at(i)(1) = itr->p1.y;
```

Comment:
Nit: assuming this works and compiles
```suggestion
            field.field_corners.at(i).x() = itr->p1.x;
            field.field_corners.at(i).y() = itr->p1.y;
```